fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/security/Ed25519/ElectionPublicKey.java

```diff
@@ -0,0 +1,133 @@
+import java.util.Objects;
+
+import ch.epfl.dedis.lib.crypto.Ed25519;
+import ch.epfl.dedis.lib.crypto.Ed25519Point;
+import ch.epfl.dedis.lib.crypto.Ed25519Scalar;
+import ch.epfl.dedis.lib.crypto.Point;
+import ch.epfl.dedis.lib.crypto.Scalar;
+import ch.epfl.dedis.lib.exception.CothorityCryptoException;
+import io.reactivex.annotations.NonNull;
+
+/**
+ * Represents a private key meant for El-Gam
+ */
+public class ElectionPublicKey {
+
+    // Point is generate with given public key
+    private final Point publicKey;
+
+    public ElectionPublicKey(@NonNull Base64URLData base64publicKey) {
+        try {
+            publicKey = new Ed25519Point(base64publicKey.getData());
+        } catch (CothorityCryptoException e) {
+            throw new IllegalArgumentException("Could not create the point for elliptic curve, please provide another key");
+        }
+    }
+
+    @Override
+    public String toString() {
+        return publicKey.toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ElectionPublicKey that = (ElectionPublicKey) o;
+        return that.getPublicKey().equals(getPublicKey());
+    }
+
+    /**
+     * @return the string encoded public key in Base64 format
+     */
+    public String encodeToBase64() {
+        Base64URLData encodedKey = new Base64URLData(publicKey.toBytes());
+        return encodedKey.getEncoded();
+    }
+
+    /**
+     * @return the public key in Base64 format (not encoded)
+     */
+    public Base64URLData toBase64() {
+        return new Base64URLData(publicKey.toBytes());
+    }
+
+    public Point getPublicKey() {
+        return publicKey;
+    }
+
+    /**
+     * Added this hash behavior by default
+     */
+    @Override
+    public int hashCode() {
+        return java.util.Objects.hash(getPublicKey().toString());
+    }
+
+    /**
+     * Encrypts with ElGamal under Ed25519 curve
+     *
+     * @param message message a 2 byte integer corresponding to the chosen vote
+     * @return Returns 2 32 byte strings which should be appended
+     * together and base64 encoded. For clarification, the first 32 bytes is point K and the second 32 bytes is point C.
+     */
+    public String encrypt(@NonNull byte[] message) {
+        if (message.length > 29) {
+            throw new IllegalArgumentException("The message should contain at maximum 29 bytes");
+        }
+        // Follows this implementation:
+        // https://github.com/dedis/cothority/blob/0299bcd78bab22bde6d6449b1594613987355535/evoting/lib/elgamal.go#L15-L23
+        // An example for embedding was found here:
+        // https://github.com/dedis/cothority/blob/0299bcd78bab22bde6d6449b1594613987355535/external/js/kyber/spec/group/edwards25519/point.spec.ts#L203
+
+        // M := cothority.Suite.Point().Embed(message, random.New())
```

Comment:
Why is there what appears to be Go code commented here ?

Reply:
Originally, the library is written in go. The Java library is just a translation of the go library. I took the examples from the go library

Reply:
My question is why did you put the go code in comments here. IMO, providing the link is sufficient

Reply:
Ohh, just that the user can compare the java and the go code, it seems more straightforward (at least for me)

Reply:
I would say that the Java programmer should not even be aware how the Go code is structured since for the developer the backend should be taken as a black box, so this Go code should be removed as Johann suggested